src/newgrf_gui.cpp

```diff
@@ -1314,10 +1320,59 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 		this->SetWidgetDisabledState(WID_NS_PRESET_SAVE, has_missing);
 	}
 
-	EventState OnKeyPress(WChar key, uint16 keycode) override
+	EventState OnKeyPressForActiveList(uint16 keycode)
 	{
-		if (!this->editable) return ES_NOT_HANDLED;
+		/* Get total actives and index of current selection. */
+		int active_pos = 0;
+		int active_count = 0;
+		for (GRFConfig *c = this->actives; c != nullptr; c = c->next, active_count++) {}
+		for (GRFConfig *c = this->actives; c != nullptr && c != this->active_sel; c = c->next, active_pos++) {}
 
+		switch (keycode) {
```

Comment:
This should use the new unified code for determining the new position for a scroll bar when these keys are pressed.

Reply:
I am not familiar with it, would you please share code pointers ?

Reply:
See https://github.com/OpenTTD/OpenTTD/commit/bf4fe19a6684231d04764c3bee5ed51e4124a925#diff-5fa9779f47ed8d1f7c513ace98f061dd5ddc230884b8b4f4e2383a5211e4275d

Reply:
Done! thank you :)